Installer fails to pick up some long-form arguments

@zbeekman, the installer fails to detect the "--branch trunk" argument in the following:
```
./install.sh -p gcc --branch trunk
```
Presumably the problem is in [parse_command_line.sh](https://github.com/sourceryinstitute/opencoarrays/blob/master/prerequisites/use-case/parse_command_line.sh).

I don't think there's any issue in parsing long arguments. I suspect that they just aren't getting passed correctly between different parts of the script, if there is an error. I confirmed that there's no issue in the parser by doing the following:

I added the info and exit statements in install.sh

```shell
### Validation (decide what's required for running your script and error out)
#####################################################################

[ -z "${LOG_LEVEL:-}" ] && emergency "Cannot continue without LOG_LEVEL. "

info "arg_p is ${arg_p:-}"

info "arg_b is ${arg_b:-}"

exit 0

# shellcheck disable=SC2154
if [[ "${arg_v}" == "${__flag_present}" || "${arg_l}" == "${__flag_present}" || ! -z "${arg_P:-${arg_U:-${arg_V:-${arg_D:-${arg_B}}}}}" ]]; then
   print_debug_only=7
   if [ "$(( LOG_LEVEL < print_debug_only ))" -ne 0 ]; then
     debug "Supressing info and debug messages: one of {-l, -v, -P, -U, -V, -D} present."
     suppress_info_debug_messages
   fi
fi
```

and this is the result:

```
517 20:59 IBB-MBP:opencoarraysCI:🍻 [master|…837⚑ 3] $ ./install.sh -p gcc --branch trunk
2016-09-12 01:01:52 UTC [     info] arg_p is gcc
2016-09-12 01:01:52 UTC [     info] arg_b is trunk
2016-09-12 01:01:52 UTC [     info] Cleaning up. Done
```

--- a/install.sh-usage
+++ b/install.sh-usage
@@ -1,4 +1,4 @@
-  -b --branch [arg]           Install the specified repository development branch.
+  -b --install-branch [arg]           Install the specified repository development branch.
   -B --list-branches [arg]    List the available branches in the specified package's repository.
   -c --with-c [arg]           Use specified C compiler. 
   -C --with-cxx [arg]         Use specified C++ compiler. 

--- a/prerequisites/build.sh
+++ b/prerequisites/build.sh
@@ -88,7 +88,7 @@ info "__os: ${__os}"
 info "__usage: ${__usage}"
 info "LOG_LEVEL: ${LOG_LEVEL}"
 
-info "-b (--branch):           ${arg_b} "
+info "-b (--install-branch):   ${arg_b} "
 info "-B (--list-branches):    ${arg_B} "
 info "-c (--with-c):           ${arg_c} "
 info "-C (--with-cxx):         ${arg_C} "

--- a/prerequisites/build.sh-usage
+++ b/prerequisites/build.sh-usage
@@ -8,8 +8,8 @@
   -f --with-fortran [arg]     Specify Fortran compiler location. Default="gfortran"
   -h --help                   This page.
   -i --install-prefix [arg]   Install package in specified path. Default="${OPENCOARRAYS_SRC_DIR%/}/prerequisites/installations/"
-  -j --num-threads [arg]      Number of threads to use when invoking make. Default="1"
   -I --install-version [arg]  Package version to install. (To see default, use -V or --print-version)
+  -j --num-threads [arg]      Number of threads to use when invoking make. Default="1"
   -l --list-packages          List the packages this script can install.
   -m --with-cmake [arg]       Use the specified CMake installation. Default="cmake"
   -M --with-mpi [arg]         Use the specified MPI installation. 

